pkg/controller/controller_ref_manager.go

```diff
@@ -350,7 +351,14 @@ func (m *ReplicaSetControllerRefManager) AdoptReplicaSet(rs *extensions.ReplicaS
 	// Note that ValidateOwnerReferences() will reject this patch if another
 	// OwnerReference exists with controller=true.
 	addControllerPatch := fmt.Sprintf(
-		`{"metadata":{"ownerReferences":[{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],"uid":"%s"}}`,
+		`{
+			"metadata":
+			{	
+				"ownerReferences": [{"apiVersion":"%s","kind":"%s","name":"%s","uid":"%s","controller":true,"blockOwnerDeletion":true}],
+				"uid":"%s",
+				"finalizers": ["foregroundDeletion"]
```

Comment:
Maybe in a followup we can make this line "finalizers": ["%s"] and supply metav1.FinalizerDeleteDependents.